operator/internal/manifests/internal/config/build_test.go

```diff
@@ -5418,3 +5418,199 @@ analytics:
 		})
 	}
 }
+
+func TestBuild_ConfigAndRuntimeConfig_STS_WIF(t *testing.T) {
+	defaultSchema := []lokiv1.ObjectStorageSchema{
+		{
+			Version:       lokiv1.ObjectStorageSchemaV11,
+			EffectiveDate: "2020-10-01",
+		},
+	}
+	for _, tc := range []struct {
+		name             string
+		objStorageConfig storage.Options
+		expStorageConfig string
+	}{
+		{
+			name: "aws_sts",
+			objStorageConfig: storage.Options{
+				SharedStore: lokiv1.ObjectStorageSecretS3,
+				S3: &storage.S3StorageConfig{
+					RoleArn:              "my-role-arn",
+					Region:               "my-region",
+					Buckets:              "my-bucket",
+				},
+				Schemas: defaultSchema,
+			},
+			expStorageConfig: `
+    s3:
+      s3: s3://my-region/my-bucket
+      s3forcepathstyle: false`,
+		},
+	} {
```

Comment:
Any reason to keep this folded as a table test? In general in this test file I would keep one test case one test function because the expected values become ugly long and table test add more reading burden.

Reply:
My goal would be to add the other test cases for the other WIF implementations but if we want to keep 1 function per test case I'm happy to oblige. Although I believe that when they are like this it's easier to read because we see all the important parts of the test at the same time.